lib/ResourceTokenizer.js

```diff
@@ -177,18 +178,36 @@ export class ResourceTokenizer {
    * Creates a new `tokenized` section for an acquisition record by
    * applying an acquire request.
    *
+   * @param {object} options - Options to use.
+   * @param {object} options.checkResults - The results from running a
+   *   resource check.
+   *
    * @returns {object} An object with `newTokenized` representing a new
-   *   `tokenized` section for an acquisition record and `expires` representing
+   *   `tokenized` section for an acquisition record, `expires` representing
    *   the earliest time that an expiration of all tracked acquisitions would
-   *   be permissible (when a full database record clean up can safely occur).
+   *   be permissible (when a full database record clean up can safely occur),
+   *   and `ttl` for more granularly expiring individual acquisitions.
    */
-  applyAcquireRequest() {
-    const {newTokenizerId, prunedRequest} = this;
+  applyAcquireRequest({checkResults} = {}) {
+    /* Compute the new TTL for the acquisition record by considering the
+    previous value and the maximum TTL from the applied restrictions. The new
+    TTL must be greater than or equal to the previous one unless it has been
+    set to `0` to indicate there are no current acquisitions. This ensures that
+    previous resource acquisitions continue to be tracked for at least as long
+    as was required when the resources were first acquired. */
+    const {maxRestrictionTtl} = checkResults;
+    const {previousAcquisitionTtl: previousTtl} = this;
+    // new TTL must be no less than the previous TTL (unless it is zero)
+    const ttl = Math.max(previousTtl || maxRestrictionTtl, maxRestrictionTtl);
```

Comment:
is it not adequate to just do?
```suggestion
    const ttl = Math.max(previousTtl, maxRestrictionTtl);
```

Reply:
At this point, I think that's fine. Previously there was a chance that `previousTtl` would be undefined. Will update.